container_files/public_html/doc/builtin/procedures/SplitProcedure.md

```diff
@@ -0,0 +1,18 @@
+# Dataset Split Procedure
+
+This procedure splits a dataset into several while respecting the distribution 
+of the presence of one or several labels.
+
+This can be used, for example, to split a dataset with multiple labels
+between training, testing and validation sets.
+
+## Algorithm
+
+The procedure uses a greedy algorithm and is not guaranteed to return the optimal solution.
+The algorithm will try to assure than every label is represented in all splits.
+If a label is represented in fewer rows than there are splits, the rows will be placed in the 
+firsts splits.
```

Comment:
Why don't we avoid the problem by recommending to pass the splits in that order test, validation then train.  This way we won't be training on labels we cannot test for.